.github/actions/login-to-azure/action.yml

```diff
@@ -0,0 +1,21 @@
+name: Login to Azure
+description: Logs in to Azure using a service principal.
+
+inputs:
+  enable-azure-powershell-session:
+    required: false
+    default: "true"
+    description: Whether to enable an Azure PowerShell session.
+  service-principal:
```

Comment:
This is a secret and should be passed around as such, in case of composite actions as an env var. See e.g. how it works with `JIRA_API_TOKEN` from the `create-jira-issues-for-community-activities` workflow.